### PR TITLE
Ed25519 key generation, signing, and verification

### PR DESCRIPTION
**Pull request by amika-sq**
This PR implements Ed25519 key generation, signing, and verification in Swift.

This PR also uses the test vectors from [web5-js](https://github.com/TBD54566975/web5-js/tree/main/packages/crypto/tests/fixtures/test-vectors/ed25519) to ensure compatibility with our other SDKs. Because of this, the PR size is quite large. But reviewers can safely skip over these test vector json files if desired to save some time, as they are just copy/pasted from that repo.

I could not find a JWK implementation in Swift that implemented both `ed25519` and `secp256k1`, and could manipulate the JWK in the same ways as we are in our SDKs. For this reason, I implemented a quick & dirty JWK representation so that we could get off the ground running here. I will be making separate backlog tasks to come back in and tidy up the JWK implementation, so that all JWKs don't have parameters that are not relevant for their keyTypes.

### CODE REVIEW

File: Sources/tbDEX/crypto/JWK.swift [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+
+    /// Supported `crv` curve types.
+    public enum Curve: String, Codable {
+        case ed25519 = "Ed25519"
+        case secp256k1 = "secp256k1"
+    }
+
+    /// Supported `kty` key types.
+    public enum KeyType: String, Codable {
+        case ecdsa = "EC"
```
Comment (decentralgabe) [inline Sources/tbDEX/crypto/JWK.swift:17]: believe this should be `case elliptic = "EC"`

File: Sources/tbDEX/crypto/JWK.swift [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+    /// The below properties represent JWK parameters that are unique to specific JWK key types.
+
+    /// `d` Private exponent.
+    public var d: String?
+
+    /// The x-coordinate for the Elliptic Curve point.
+    public var x: String?
+
+    /// Elliptic Curve y-coordinate.
+    public var y: String?
```
Comment (decentralgabe) [inline Sources/tbDEX/crypto/JWK.swift:98]: is there a JWK lib we could use instead here? there are other JWK properties that might show up

maybe https://github.com/airsidemobile/JOSESwift

**Review by decentralgabe — APPROVED** [review]
nice!

**Review by mistermoe — APPROVED** [review]

File: Sources/tbDEX/crypto/JWK.swift (hunk 2 above)
Comment (amika-sq) [reply Sources/tbDEX/crypto/JWK.swift:98]: I opted not to do this, as our usage for JWKs in tbDEX is very limited, and these libraries are much larger than we actually need. On top of that, they've modeled certain aspects of their implementation with enums, which we cannot extend as they are closed types. 

For these reasons, I've opted to make a super minimal representation of JWK for now. We can always revisit in the future.

**Pull request merged**